test/local/DSL/limitSwapAction.test.ts

```diff
@@ -0,0 +1,55 @@
+import { expect } from 'chai';
+import { limitSwapAction } from '../../../src'; // Assuming your implementation is in this path
+
+const USDC_ADDRESS = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48';
+const WETH_ADDRESS = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2';
+
+describe('limitSwapAction', function () {
+
+  it('should handle token swap with all expected attributes', function () {
+    const primitives = limitSwapAction({
+      type: 'limitSwap',
+      id: BigInt('1'),
+      owner: '0xOwnerAddress',
+      tokenIn: USDC_ADDRESS,
+      tokenOut: WETH_ADDRESS,
+      tokenInAmount: 1000000,
+      tokenOutAmount: 1
+    });
+
```

Comment:
might be extra since the other fns are tested, but could check the price curve params here and test for the priceCurveParams data to be hex value of 1/1000000 * 2**96 converted to hex

also could check that it has `type: 'limitSwapExactInput'`